### PR TITLE
Add FinalDBSnapshotIdentifier property to RDS delete instance action

### PR DESCRIPTION
**Pull request by ravikiranvm**
Fixes OPS-2300.

## Additional Notes

Used dynamic property for "Snapshot Identifier" to conditionally display depending on the "Take Snapshot" toggle

Screenshot 2025-08-29 at 4 13 24 PM

Screenshot 2025-08-29 at 4 01 14 PM

## Testing Checklist

Check all that apply:

- [ ] I tested the feature thoroughly, including edge cases

- [ ] I verified all affected areas still work as expected

- [x] Automated tests were added/updated if necessary

- [x] Changes are backwards compatible with any existing data, otherwise a migration script is provided

### CODE REVIEW

File: packages/openops/src/lib/aws/rds/rds-delete-instance.ts [hunk 1]
```diff
@@ -10,13 +10,20 @@ export async function initiateRdsInstanceDeletion(
...
   try {
-    const command = new RDS.DeleteDBInstanceCommand({
+    const commandInput: RDS.DeleteDBInstanceCommandInput = {
       DBInstanceIdentifier: instanceId,
       SkipFinalSnapshot: !createSnapshot,
-    });
+    };
+
+    if (createSnapshot && finalSnapshotIdentifier) {
+      commandInput.FinalDBSnapshotIdentifier = finalSnapshotIdentifier;
```
[the remark this answers is not in the document]
Comment (ravikiranvm) [reply packages/openops/src/lib/aws/rds/rds-delete-instance.ts:23]: I think it is better to let users give a unique identifier rather than generating one.

File: packages/blocks/aws/src/lib/actions/rds/rds-delete-instance-action.ts [hunk 2]
```diff
@@ -42,12 +52,21 @@ export const rdsDeleteInstanceAction = createAction({
     const waitForInSeconds =
       waitForTimeInSecondsProperty['waitForTimeInSeconds'];
     const credentials = await getCredentialsForAccount(context.auth, accountId);
+
+    let finalSnapshotIdentifier: string | undefined = undefined;
+    if (takeSnapshot) {
+      if (snapshotIdentifier && snapshotIdentifier.trim().length > 0) {
+        finalSnapshotIdentifier = snapshotIdentifier.trim();
+      }
+    }
```
[the remark this answers is not in the document]
Comment (ravikiranvm) [reply packages/blocks/aws/src/lib/actions/rds/rds-delete-instance-action.ts]: Done

**Review by MarceloRGonc — CHANGES_REQUESTED** [review]
The property should only appear if the 'Take Snapshot' toggle is active. You can use a Dynamic Property if necessary. We have more examples like this for you to see how it works.

File: packages/openops/src/lib/aws/rds/rds-delete-instance.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ravikiranvm) [reply packages/openops/src/lib/aws/rds/rds-delete-instance.ts:23]: Used dynamic property instead

**Comment by ravikiranvm** [comment]
> The property should only appear if the 'Take Snapshot' toggle is active. You can use a Dynamic Property if necessary. We have more examples like this for you to see how it works.

Done, thank you.

**Review by MarceloRGonc — APPROVED** [review]
